config/initializers/secure_headers.rb

```diff
@@ -33,3 +33,12 @@
   #   ]
   # }
 end
+
+SecureHeaders::Configuration.override(:saml) do |config|
+  providers = YAML.load_file("#{Rails.root}/config/service_providers.yml")
```

Comment:
I verified this by adding a `puts` statement in the Secure Headers config, and verifying that it was only called once, even though the SamlIdpController action was called several times.